### PR TITLE
Rewrite of InLayerEdgeTwoNodeCrossingCounter

### PR DESCRIPTION
**Pull request by altre**
Signed-off-by: Alan Schelten a.schelten@gmail.com

### CODE REVIEW

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/greedyswitch/InLayerEdgeTwoNodeCrossingCounter.java [hunk 1]
```diff
@@ -10,178 +10,312 @@
...
     /**
-     * Counts crossings between in-layer edges incident to two nodes.
+     * Create crossings counter.
      * 
-     * @param nodeOrder
-     *            the current order of the layer to be counted in.
+     * @param side
+     * 
+     * @param portPositions
+     *            port position array passed to prevent frequent large array construction.
```
Comment (le-cds) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/greedyswitch/InLayerEdgeTwoNodeCrossingCounter.java]: At least have the decency to document all the parameters. ;)

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/greedyswitch/InLayerEdgeTwoNodeCrossingCounter.java [hunk 2]
```diff
@@ -10,178 +10,312 @@
...
-     * 
-     * @param upper
-     *            the upper node
-     * @param lower
-     *            the lower node
+     * Count crossings only between two nodes.
+     *
+     * @param upperNode
+     * @param lowerNode
+     * @param side
```
Comment (le-cds) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/greedyswitch/InLayerEdgeTwoNodeCrossingCounter.java]: This is not a parameter anymore, the other two should be documented.

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/greedyswitch/InLayerEdgeTwoNodeCrossingCounter.java [hunk 3]
```diff
@@ -10,178 +10,312 @@
...
+     * @return
```
Comment (le-cds) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/greedyswitch/InLayerEdgeTwoNodeCrossingCounter.java]: Remove.

File: plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/greedyswitch/InLayerEdgeTwoNodeCrossingCounter.java [hunk 4]
```diff
@@ -1,5 +1,5 @@
 /*******************************************************************************
- * Copyright (c) 2014, 2015 Kiel University and others.
+ * Copyright (c) 2016 Kiel University and others.
```
Comment (eNBeWe) [inline plugins/org.eclipse.elk.alg.layered/src/org/eclipse/elk/alg/layered/intermediate/greedyswitch/InLayerEdgeTwoNodeCrossingCounter.java]: I think 2014, 2016 would be more appropriate, even when only about 10 lines of code remain from the previous implementation.

**Comment by eNBeWe** [comment]
@altre Could you do these minor changes and commit these? I would like to have this pull request merged before the next release ;)

**Comment by altre** [comment]
Should I really? I thought the verdict was to integrate the big change for the hierarchical graphs. That should include this change.

**Pull request closed without merging**